agent/php_rinit.c

```diff
@@ -96,6 +96,17 @@ PHP_RINIT_FUNCTION(newrelic) {
       "(^([a-z_-]+[_-])([0-9a-f_.]+[0-9][0-9a-f.]+)(_{0,1}.*)$|(.*))",
       NR_REGEX_CASELESS, 0);
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+    && !defined OVERWRITE_ZEND_EXECUTE_DATA
+  /* Stack needs to have a dtor set so that when we free it
+   * during rshutdown, all elements are properly freed */
+  void str_stack_dtor(void* e, NRUNUSED void* d) {
+    char* str = (char*)e;
+    nr_free(str);
+  }
+  nr_stack_init(&NRPRG(predis_ctxs), NR_STACK_DEFAULT_CAPACITY);
```

Comment:
Thanks for working on this.
Similar question to the other PR.  Does this need deinit or is destroy enough?

Reply:
The dtor is set separately from the stack_init method, on the next line. Ideally, the stack should never have anything in it at the end of a transaction. However, by setting the dtor, we ensure that any unforeseen situations do not result in a memory leak if the stack is still populated on destruction.